src/Code.js

```diff
@@ -58,27 +38,47 @@ function setApiKey() {
   const button = result.getSelectedButton();
   const text = result.getResponseText();
   if (button === ui.Button.OK) {
-    setProperty(PROP_MB_API_KEY, text);
-    ui.alert(`API key is ${text}.`);
+    try {
+      setProperty(PROP_MB_API_KEY, text);
+      ui.alert(`API key is ${text}`);
+    } catch (e) {
+      ui.alert(e.message);
+    }
   }
 }
 
 function deleteAllSettings() {
-  const properties = PropertiesService.getDocumentProperties();
-  if (properties) {
-    properties.deleteAllProperties();
-  } else {
-    testProperties = {};
+  const ui = SpreadsheetApp.getUi();
+  const result = ui.alert(
+    'Are you sure to delete all settings?',
+    ui.ButtonSet.YES_NO,
+  );
+
+  if (result === ui.Button.YES) {
+    deleteAllProperties();
+    ui.alert('Deployment ID and API key have been deleted');
   }
-  SpreadsheetApp.getUi()
-    .alert('Deployment ID and API key have been removed.');
 }
 
 function refreshCurrentCell() {
+  const ui = SpreadsheetApp.getUi();
+
+  if (!credentialsExist()) {
+    ui.alert('Credentials are invalid. Please see https://www.curvegrid.com/multibaas-for-google-sheets/credentials');
```

Comment:
These URLs don't exist yet - let's chat about how to handle them/where to put them (docs vs. landing page, etc.).

Reply:
Yeah, I am thinking of using https://github.com/curvegrid/v2.curvegrid.com/pull/198 after merged but we can chat later. :)